test/unit/specs/components/v-button.spec.js

```diff
@@ -0,0 +1,125 @@
+
+/**
+ * Throughout this suite we use the `screen.findBy*` functions to asynchronously
+ * wait for the component to re-render. There might be some kind of performance
+ * problem with the component's implementation, but if we don't "wait" for it
+ * to settle, then all the props that are changed after `onMounted` completes
+ * won't be rendered.
+ */
+describe('VButton', () => {
+  it('should render a `button` by default with type="button" and no tabindex', async () => {
+    render(VButton, {
+      slots: { default: 'Code is Poetry' },
+    })
+
+    const element = await screen.findByRole('button')
+
+    expect(element.tagName).toBe('BUTTON')
+    expect(element).toHaveAttribute('type', 'button')
+    expect(element).not.toHaveAttribute('tabindex')
+  })
+
+  it('should allow passing an explicit type', async () => {
+    render(VButton, {
+      props: { type: 'submit' },
+      slots: { default: 'Code is Poetry' },
+    })
+
+    const element = await screen.findByRole('button')
+
+    expect(element).toHaveAttribute('type', 'submit')
+  })
+
+  it('should allow passing in an explicit tabindex', async () => {
+    render(VButton, {
+      props: { tabindex: '-1' },
+      slots: { default: 'Code is Poetry' },
+    })
+
+    const element = await screen.findByRole('button')
+
+    expect(element).toHaveAttribute('tabindex', '-1')
+  })
+
+  it.each(['button', 'submit', 'reset', 'color', 'file', 'image'])(
+    'should allow an input element that renders a button with the type %s',
+    async (type) => {
+      const TestWrapper = Vue.component('TestWrapper', {
+        components: { VButton },
+        template: `<label>Code is Poetry<VButton as="input" type="${type}" /></label>`,
+      })
+      render(TestWrapper)
+
+      const element = await screen.findByLabelText(/code is poetry/i)
+
+      expect(element.tagName).toBe('INPUT')
+      expect(element).toHaveAttribute('type', type)
+    }
+  )
+
+  // @todo(sarayourfriend) fix this failing test!
+  it.skip('should render an anchor with no type attribute', async () => {
+    render(VButton, {
+      props: { as: 'a' },
+      slots: { default: 'Code is Poetry' },
+    })
+
+    const element = await screen.findByText(/code is poetry/i)
+
+    screen.debug(element)
+
+    expect(element.tagName).toBe('A')
+    expect(element).not.toHaveAttribute('type')
+  })
+
+  it('should render as a div with role and tabindex', async () => {
+    render(VButton, {
+      props: { as: 'div' },
+      slots: { default: 'Code is Poetry' },
+    })
+
+    const element = await screen.findByRole('button')
+
+    expect(element.tagName).toBe('DIV')
+    expect(element).toHaveAttribute('role', 'button')
+    expect(element).toHaveAttribute('tabindex', '0')
+  })
+
+  it('should render the disabled attribute on a button when the element is explicitly unfocusable and is disabled', async () => {
+    render(VButton, {
+      props: { disabled: true, focusable: false },
```

Comment:
I don't understand why setting a button 'disabled' doesn't also make it unfocusable? It seems to do so when I add `disabled="disabled"` to the Openverse Search button: I can't tab there...

Reply:
There are cases where you would want to render a button as disabled but still focusable, for example, the "submit" button of a form. You want there to still be a submit form that the screen reader is capable of navigating to but you want to communicate that it is disable due to conditions of the form (for example, required fields not being filled out or fields being filled out with invalid input).

`disabled="disabled"` makes the button completely ignored by the screen reader, meaning you would not be able to convey that state that users who can see the button will get to users who cannot see the button.